woocommerce/woocommerce
# Remove Selenium e2e tests & add Puppeteer new product e2e test

### All Submissions:






### Changes proposed in this Pull Request:



1) Improvements to Puppeteer's e2e test infrastructure:

- Add `--verbose` argument for running new Puppeteer e2e tests. `--verbose` allows seeing the status of each test step within a test once the test is completed:

![verbose](https://user-images.githubusercontent.com/19143190/63771134-77308f00-c8ce-11e9-8c6d-1a1eb8aa12ce.jpg)

- Add `Debugging tests` chapter in e2e tests README file;
- Add the `debugger` option in order to debug tests.

2) Remove Selenium-based e2e tests;

- Remove `wp-admin` and `front-end` e2e tests. New Puppeteer tests will be written using the same model. 
- Remove `wc-e2e-page-objects` package used for writing Selenium based e2e tests;

3) Add new Puppeteer e2e test: 

- New product in wp-admin e2e test;
- Add a new file with `flows` needed to run the tests;
- Add a new file with helper functions needed to run tests. 

### How to test the changes in this Pull Request:

1. Setup e2e test environment as per README [here](https://github.com/woocommerce/woocommerce/pull/24218/files);
2. Enable `Development mode` by passing `--dev` flag to the `test:e2e` script in `./package.json` file as follows: `"test:e2e": "./tests/bin/e2e-test-integration.js --dev",`
3. Run e2e test by using `npm run test:e2e` command;
4. Test `debugger` by placing `await jestPuppeteer.debug();` anywhere in test code to pause execution. Run `npm run test:e2e` command. You should see the following once test execution stops:

![debugger](https://user-images.githubusercontent.com/19143190/63772633-67667a00-c8d1-11e9-9f22-632ecd08ccb2.jpg)

You should also see an indication that test execution had been stopped in the terminal:

![terminal_debugger](https://user-images.githubusercontent.com/19143190/63772880-f378a180-c8d1-11e9-9752-ece7b4095ebb.jpg)

5. Test headless by disabling `Development mode` and running `npm run test:e2e` command. 

### Note:

Credit for writing the test goes to @jeffstieler. On my end, I am breaking down his PR (https://github.com/woocommerce/woocommerce/pull/23548) into chunks for review and introduce minor improvements along the way. 

### Other information:




### Changelog entry

> Remove Selenium e2e tests & add Puppeteer new product e2e test
